pkg/operator/sync_v310_00.go

```diff
@@ -23,6 +23,100 @@ import (
 
 // most of the time the sync method will be good for a large span of minor versions
 func sync_v310_00_to_00(c ServiceCertSignerOperator, operatorConfig *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity) (operatorsv1alpha1.VersionAvailablity, []error) {
+	signingVersionAvailability, signingErrors := sync_v310_00_to_00_SigningController(c, operatorConfig, previousAvailability)
+	apiServiceInjectorVersionAvailability, apiServiceInjectorErrors := sync_v310_00_to_00_APIServiceController(c, operatorConfig, previousAvailability)
+
+	allErrors := []error{}
+	allErrors = append(allErrors, signingErrors...)
+	allErrors = append(allErrors, apiServiceInjectorErrors...)
+
+	mergedVersionAvailability := operatorsv1alpha1.VersionAvailablity{
+		Version: operatorConfig.Spec.Version,
+	}
+	mergedVersionAvailability.Generations = append(mergedVersionAvailability.Generations, signingVersionAvailability.Generations...)
+	mergedVersionAvailability.Generations = append(mergedVersionAvailability.Generations, apiServiceInjectorVersionAvailability.Generations...)
+	if signingVersionAvailability.UpdatedReplicas > 0 && apiServiceInjectorVersionAvailability.UpdatedReplicas > 0 {
+		mergedVersionAvailability.UpdatedReplicas = 1
+	}
+	if signingVersionAvailability.ReadyReplicas > 0 && apiServiceInjectorVersionAvailability.ReadyReplicas > 0 {
+		mergedVersionAvailability.ReadyReplicas = 1
+	}
+
+	return mergedVersionAvailability, allErrors
+}
+
+func sync_v310_00_to_00_APIServiceController(c ServiceCertSignerOperator, operatorConfig *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity) (operatorsv1alpha1.VersionAvailablity, []error) {
+	versionAvailability := operatorsv1alpha1.VersionAvailablity{
+		Version: operatorConfig.Spec.Version,
+	}
+
+	errors := []error{}
+	var err error
+
+	requiredNamespace := resourceread.ReadNamespaceV1OrDie(v310_00_assets.MustAsset("v3.10.0/apiservice-cabundle-controller/ns.yaml"))
+	_, _, err = resourceapply.ApplyNamespace(c.corev1Client, requiredNamespace)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "ns", err))
```

Comment:
wondering whether it is a good idea to continue modifying everything else after the first error. This is risky, isn't it?

Reply:
> wondering whether it is a good idea to continue modifying everything else after the first error. This is risky, isn't it?

We should be able to continue safely since we own the objects.  I think that continuing on to reset objects to our view of the world is better than letting them remain corrupted.